deps/build.jl

```diff
@@ -0,0 +1,84 @@
+# Builds a GAP, and attaches JuliaInterface
+
+import Pkg
+
+julia_folder_path = abspath(joinpath(@__DIR__,".."))
+
+gap_root = abspath(joinpath(julia_folder_path,"gap"))
+install_gap = true
+
+if haskey(ENV,"GAPROOT")
+    gap_root = ENV["GAPROOT"]
+    install_gap = false
+end
+
+## Find julia binary
+julia_binary = julia_binary = get(ENV, "JULIA_BINARY", Sys.BINDIR)
+
+## Install GAP
+if install_gap
+    cd(julia_folder_path)
+    run(`git clone https://github.com/gap-system/gap`)
+    cd(gap_root)
+    run(`./autogen.sh`)
+    run(`./configure --with-gc=julia --with-julia=$(julia_binary)`)
+    run(`make`)
+    if ! haskey(ENV,"GAP_INSTALL_PACKAGES") || ENV["GAP_INSTALL_PACKAGES"] == "yes" 
+        run(`make bootstrap-pkg-full`)
+        cd("pkg")
+        run(`../bin/BuildPackages.sh`)
```

Comment:
Why not?

Reply:
Because what it does is hard to predict; every use will potentially get a slightly different set of packages that work, making this into a bit of a support nightmare. Of course that's a deeper issue, and one of the many reasons `BuildPackages.sh` is only a stop-gap measure, but not a complete solution for the GAP package management issue. Anyway, leave it as is, it's just one more spot we'll have to change once `BuildPackages.sh` gets replaced by something better.

Reply:
Okay, then let's change this once we have the package manager.